### ISSUE TITLE
Issues when uploading certain files

### ISSUE THREAD

**Issue by jwahsnakupaku** [target/strelka-ui#48]
When trying to upload the Emotet sample file in the doc it silently fails, appears to upload but the results don't appear.
If I unzip the sample it works fine.

Tried a few other types, something as simple as this doesn't work;
echo lol > lol.txt

No dramas when using the strelka-oneshot client.

**Comment by phutelmyer** [comment]
Thank you for reporting this @jwahsnakupaku. I am aware of this issue and am actively looking into a fix. I will comment here when I implement a fix.

**Comment by jwahsnakupaku** [comment]
Spun up a dev environment to  have a little look.
Looks like it is an issue when there are no yara matches, if I remove some try excepts to force the error;
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/app/services/auth.py", line 81, in decorated_function
    return f(user, *args, **kwargs)
  File "/app/blueprints/strelka.py", line 112, in submit_file
    get_yara_hits(response),
  File "/app/blueprints/strelka.py", line 227, in get_yara_hits
    for yarahit in r["scan"]["yara"]["matches"]:
KeyError: 'matches'
```

With lol.txt there'll be no matches, in r['scan'] there is only;
` 'yara': {'elapsed': 0.089633}}`

Can do something like this;
```
def get_yara_hits(response: list) -> list:
    """
    Get a list of unique YARA rules that matched the file in the Strelka response.

    Args:
        response (list): The response from Strelka.

    Returns:
        list: A list of YARA rule names.
    """
    yarahits = set()
    try:
        for r in response:
            for yarahit in r["scan"]["yara"]["matches"]:
                yarahits.add(yarahit)
    except KeyError as e:
        pass
    return list(yarahits)
```
But, the UI still expects it to be there when displaying and throws some JS error, presumably because it is displaying it at the top summary.

Could either dump something like this in there, feels pretty dodgy though;
```
            for r in response:
                if 'matches' not in r["scan"]["yara"]:
                    r["scan"]["yara"]["matches"] = []
            # Create a new submission object and add it to the database.
            new_submission = FileSubmission(
```

Or maybe Strelka should return an empty list for matches if nothing is found.

**Comment by phutelmyer** [comment]
@jwahsnakupaku,

Great observation - that is definitely an oversight, and I was able to recreate the bug. I fixed this specific bug [here](https://github.com/target/strelka-ui/pull/47/commits/e1c071de2264dc5e3b955d61cd8095bbed1b1ec8). 

Fixed in #47
